corehq/apps/analytics/static/analytics/js/kissmetrics.js

```diff
@@ -0,0 +1,152 @@
+/* globals _, _kmq */
+
+var _kmq = _kmq || [];
+
+hqDefine('analytics/js/kissmetrics', function () {
+    'use strict';
+    var _get = hqImport('analytics/js/initial').getFn('kissmetrics'),
+        _abTests = hqImport('analytics/js/initial').getAbTests('kissmetrics'),
+        logger = hqImport('analytics/js/logging').getLoggerForApi('Kissmetrics'),
+        _utils = hqImport('analytics/js/utils'),
+        _init = {};
+
+    logger.verbose.addCategory('data', 'DATA');
+    logger.debug.addCategory('ab', 'AB TEST');
+
+    window.dataLayer = window.dataLayer || [];
+
+    var KmqWrapper = function (originalObject) {
```

Comment:
I'm not totally following the purpose of `KmqWrapper`, is it just that it logs when deprecated code is used?

Reply:
it logs when `_kmq.push` is called directly